pkg/component/etcd/etcd.go

```diff
@@ -546,9 +561,13 @@ func (e *etcd) Deploy(ctx context.Context) error {
 			return err
 		}
 	} else {
+		// Neither VPA nor HVPA is enabled for etcd, delete the remaining objects
 		if err := kubernetesutils.DeleteObjects(ctx, e.client, hvpa); err != nil {
 			return err
 		}
+		if err = kubernetesutils.DeleteObjects(ctx, e.client, vpa); err != nil {
+			return err
+		}
```

Comment:
Ah, I looked at the tests you referenced and turns out, they're dealing with the `Destroy` method for etcd. This method _was_ missing the VPA component, so that was a good catch!